File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index = []
        for i in range(0,len(nums)):
            indLen = len(index)
            for j in range(0+1,len(nums)):
                if nums[i] + nums[j] == target and indLen < 2 and j != i:
                    index.append(i)
                    index.append(j)
        return list(set(index))
            
         
        

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> numMap; // Map to store (target - nums[i]) and its index

        for (int i = 0; i < nums.size(); ++i) {
            int complement = target - nums[i];

            // Check if the complement is already in the map
            if (numMap.find(complement) != numMap.end()) {
                // Return the indices of the two numbers
                return {numMap[complement], i};
            }

            // Store the current number and its index in the map
            numMap[nums[i]] = i;
        }

        // Return an empty vector if no solution is found
        return {};
        
    }
};